lib/SIL/SILType.cpp

```diff
@@ -638,17 +638,17 @@ bool SILModuleConventions::isPassedIndirectlyInSIL(SILType type, SILModule &M) {
 }
 
 bool SILFunctionType::isNoReturnFunction() {
-  return getDirectFormalResultsType().getSwiftRValueType()->isUninhabited();
+  if (getNumResults() == 1)
+    return getSingleResult().getType()->isUninhabited();
```

Comment:
A function would be noreturn if *any* of its results are uninhabited, wouldn't it?